titan-server/src/organizations/organizations.rs

```diff
@@ -30,6 +43,102 @@ pub fn find_users(
     users_query.load::<(models::WcfUser, models::WcfUserAvatar)>(&**unkso_main)
 }
 
+/// Returns all organizations.
 pub fn find_all(titan_primary: TitanPrimary) -> QueryResult<Vec<models::Organization>> {
     schema::organizations::table.load::<models::Organization>(&*titan_primary)
 }
+
+/// Returns the users belonging to an organization.
+pub fn find_users(
+    id: i32, children: bool, titan_db: &MysqlConnection)-> QueryResult<Vec<models::User>> {
+    let child_ids =
+        if children { find_children_ids_recursive(id, titan_db) } else { vec![id] };
+
+    schema::users::table.inner_join(schema::organizations_users::table)
+        .select(schema::users::all_columns)
+        .filter(schema::organizations_users::organization_id.eq_any(child_ids))
+        .order_by(schema::users::username.desc())
+        .load::<(models::User)>(titan_db)
+}
+
+/// Returns true if a user is a non-leadership member of an
+/// organization.
+///
+/// The user's role in the organization is not checked. Will return
+/// true if the user is a member of a disabled organization.
+pub fn is_user_organization_member(org_id: i32, user_id: i32, titan_db: &MysqlConnection) -> bool {
+    let organization = schema::organizations_users::table
+        .filter(schema::organizations_users::organization_id.eq(org_id))
+        .filter(schema::organizations_users::user_id.eq(user_id))
+        .first::<(models::OrganizationUser)>(titan_db);
+
+    organization.is_ok()
+}
+
+/// Find all organizations where a user is a member, but does not
+/// hold a leadership position.
+///
+/// Disabled organizations are excluded from the results.
+pub fn find_all_by_user(
+    user_id: i32,
+    titan_db: &MysqlConnection
+) -> QueryResult<Vec<models::UserOrganizationMembership>> {
+    let mut orgs = schema::organizations_users::table
+        .inner_join(schema::organizations::table)
+        .select(schema::organizations::all_columns)
+        .filter(schema::organizations::is_enabled.eq(true))
+        .filter(schema::organizations_users::user_id.eq(user_id))
+        .get_results::<(models::Organization)>(titan_db)?;
+
+    let mut memberships: Vec<models::UserOrganizationMembership> = vec!();
+    for org in orgs.drain(..) {
+        memberships.push(models::UserOrganizationMembership {
+            organization: org,
+            role: None
+        });
+    }
+
+    Ok(memberships)
+}
+
+/// Recursively queries the children of an organization and returns
+/// their IDs.
+///
+/// The list of IDs will be ordered from closest relative to most
+/// distant.
+///
+/// # Example
+/// Consider the following hierarchy of organizations, where "(n)" is
+/// an organization's ID.
+///
+///     Coast Guard (0)
```

Comment:
The indentation here is being interpreted as a code block (as per the Markdown spec), so Cargo is trying to compile and run it through doc tests. This doesn't work since it's not actually a block of code. If you want to keep the indentation, you'll need to wrap it in a fenced code block and tell Cargo to ignore it, like so:

> \`\`\`ignore
> // Whatever is here gets ignored when `cargo test` runs doc tests
> \`\`\`